client/src/features/calendar/Calendar.js

```diff
@@ -39,17 +39,30 @@ const Calendar = ({ date }) => {
     const fetch = async () => {
       // Database data from server
       const response = await DataService.getAll();
+      // if you want to test the loading behavior
+      // await new Promise(resolve => setTimeout(resolve, 20_000));
       setEvents(response.data);
     };
 
-    fetch().then(setLoading(false)).catch(setLoading(false));
+    fetch()
+      .then(() => {
+        setLoading(false);
+      })
+      .catch(e => {
+        alert(e);
+        setLoading(false);
+      });
   }, [setEvents]);
 
-  // Render nothing while fetching for data from server
-  if (loading) return null;
+  // while we are loading events, add the "animate-pulse" class to show skeleteon loading effect
```

Comment:
concatenate the Tailwind `animate-pulse` class to the containing `div` while the data is loading.